packages/flame/lib/src/components/component.dart

```diff
@@ -23,35 +25,38 @@ class Component {
   /// to the root `FlameGame`.
   PositionType positionType = PositionType.game;
 
-  /// Whether this component has been prepared and is ready to be added to the
-  /// game loop.
-  bool isPrepared = false;
+  LifecycleState _state = LifecycleState.uninitialized;
```

Comment:
```suggestion
  var _state = LifecycleState.uninitialized;
```

Reply:
Isn't this something we can enforce using lint rules?

Reply:
Like the [omit_local_variable_type](https://dart-lang.github.io/linter/lints/omit_local_variable_types.html)

Reply:
Yeah, I thought we already had that rule, but apparently not

Reply:
We have that rule, but I don't think this variable counts as local since it's a top-level class variable, even though it is private?

Reply:
That is true, so we need a different rule for this?

Reply:
Isn't it good to have the type on top-level since it is living in a bit larger context? And then it will be consistent with public class variables too.

Reply:
Yeah true true, I'm still stuck in the mindset that it is nice to see it directly to the left.

Reply:
Either way, this is just a nit, totally not a blocker.